runtime/common/ArgumentConversion.cpp

```diff
@@ -99,12 +99,245 @@ static Value genConstant(OpBuilder &, cudaq::cc::StructType, void *,
 static Value genConstant(OpBuilder &, cudaq::cc::ArrayType, void *,
                          ModuleOp substMod, llvm::DataLayout &);
 
+/// Create callee.init_N that initializes the state
+/// Callee:
+/// func.func @__nvqpp__mlirgen__callee(%arg0: i64) {
+///   %0 = cc.alloca i64
+///   cc.store %arg0, %0 : !cc.ptr<i64>
+///   %1 = cc.load %0 : !cc.ptr<i64>
+///   %2 = quake.alloca !quake.veq<?>[%1 : i64]
+///   %3 = quake.extract_ref %2[1] : (!quake.veq<?>) -> !quake.ref
+///   quake.x %3 : (!quake.ref) -> ()
+///   return
+/// }
+/// callee.init_N:
+/// func.func private @callee.init_0(%arg0: !quake.veq<?>, %arg0: i64) ->
+/// !!quake.veq<?> {
+///   %1 = quake.extract_ref %arg0[1] : (!quake.veq<2>) -> !quake.ref
+///   quake.x %1 : (f64, !quake.ref) -> ()
+///   return %arg0: !quake.veq<?>
+/// }
+static void createInitFunc(OpBuilder &builder, ModuleOp sourceMod,
+                           func::FuncOp calleeFunc,
+                           std::string &initKernelName) {
+  OpBuilder::InsertionGuard guard(builder);
+  builder.setInsertionPointToEnd(sourceMod.getBody());
+
+  auto ctx = builder.getContext();
+  auto loc = builder.getUnknownLoc();
+
+  auto initFunc = cast<func::FuncOp>(builder.clone(*calleeFunc));
+
+  auto argTypes = calleeFunc.getArgumentTypes();
+  auto retTy = quake::VeqType::getUnsized(ctx);
+  auto funcTy = FunctionType::get(ctx, argTypes, TypeRange{retTy});
+
+  initFunc.setName(initKernelName);
+  initFunc.setType(funcTy);
+  initFunc.setPrivate();
+
+  OpBuilder newBuilder(ctx);
+
+  auto *entryBlock = &initFunc.getRegion().front();
+  newBuilder.setInsertionPointToStart(entryBlock);
+  auto intType = newBuilder.getI64Type();
+  Value zero = newBuilder.create<arith::ConstantIntOp>(loc, 0, intType);
+  Value one = newBuilder.create<arith::ConstantIntOp>(loc, 1, intType);
```

Comment:
```suggestion
  Value zero = newBuilder.create<arith::ConstantIntOp>(loc, 0, 64);
  Value one = newBuilder.create<arith::ConstantIntOp>(loc, 1, 64);
```
nit: I've been hardcoding the size as an integer everywhere.